### ISSUE TITLE
bug: SSL options in bentoml_configuration.yaml are ignored

### ISSUE THREAD

**Issue by ryanheise** [bentoml/BentoML#3570]
### Describe the bug

The `ssl` section in `bentoml_configuration.yaml` is ignored.

### To reproduce

Create a self-signed certificate:

```bash
openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout key.key -out cert.crt
```

Create the file `bentoml_configuration.yaml` as follows:

```yaml
api_server:
  ssl:
    enabled: true
    certfile: cert.crt
    keyfile: key.key
```

Run your project with the config:

```bash
BENTOML_CONFIG=bentoml_configuration.yaml bentoml serve service:svc --production --verbose
```

THE RESULT:

```
2023-02-15T18:32:29+1100 [DEBUG] [cli] Importing service "service:svc" from working dir: "/home/ryan/whatever/bentoml/src"
2023-02-15T18:32:30+1100 [DEBUG] [cli] 'transcription_pipeline' imported from source: bentoml.Service(name="transcription_pipeline", import_str="service:svc", working_dir="/home/ryan/whatever/bentoml/src")
2023-02-15T18:32:30+1100 [DEBUG] [cli] Runner map: {}
2023-02-15T18:32:30+1100 [INFO] [cli] Prometheus metrics for HTTP BentoServer from "service:svc" can be accessed at http://localhost:3000/metrics.
2023-02-15T18:32:31+1100 [INFO] [cli] Installing handle_callback_exception to loop
2023-02-15T18:32:31+1100 [INFO] [cli] Registering signals...
2023-02-15T18:32:31+1100 [INFO] [cli] Starting master on pid 2603823
2023-02-15T18:32:31+1100 [DEBUG] [cli] Socket bound at 0.0.0.0:3000 - fd: 3
2023-02-15T18:32:31+1100 [INFO] [cli] sockets started
2023-02-15T18:32:31+1100 [DEBUG] [cli] Initializing watchers
2023-02-15T18:32:31+1100 [DEBUG] [cli] cmd: /usr/bin/python
2023-02-15T18:32:31+1100 [DEBUG] [cli] args: ['-m', 'bentoml_cli.worker.http_api_server', 'service:svc', '--fd', '$(circus.sockets._bento_api_server)', '--runner-map', '{}', '--working-dir', '/home/ryan/whatever/bentoml/src', '--backlog', '2048', '--worker-id', '$(CIRCUS.WID)', '--prometheus-dir', '/home/ryan/bentoml/prometheus_multiproc_dir']
2023-02-15T18:32:31+1100 [DEBUG] [cli] process args: ['/usr/bin/python', '-m', 'bentoml_cli.worker.http_api_server', 'service:svc', '--fd', '3', '--runner-map', '{}', '--working-dir', '/home/ryan/whatever/bentoml/src', '--backlog', '2048', '--worker-id', '1', '--prometheus-dir', '/home/ryan/bentoml/prometheus_multiproc_dir']
2023-02-15T18:32:31+1100 [DEBUG] [cli] running api_server process [pid 2603868]
2023-02-15T18:32:31+1100 [INFO] [cli] Arbiter now waiting for commands
2023-02-15T18:32:31+1100 [INFO] [cli] api_server started
2023-02-15T18:32:31+1100 [INFO] [cli] Starting production HTTP BentoServer from "service:svc" listening on http://0.0.0.0:3000 (Press CTRL+C to quit)
2023-02-15T18:32:32+1100 [DEBUG] [api_server:1] Importing service "service:svc" from working dir: "/home/ryan/whatever/bentoml/src"
2023-02-15T18:32:33+1100 [DEBUG] [api_server:1] 'transcription_pipeline' imported from source: bentoml.Service(name="transcription_pipeline", import_str="service:svc", working_dir="/home/ryan/whatever/bentoml/src")
2023-02-15T18:32:33+1100 [DEBUG] [api_server:1] 'tracing.sample_rate' is set to zero. No traces will be collected. Please refer to https://docs.bentoml.org/en/latest/guides/tracing.html for more details.
2023-02-15T18:32:33+1100 [INFO] [api_server:1] Started server process [2603868]
2023-02-15T18:32:33+1100 [INFO] [api_server:1] Waiting for application startup.
2023-02-15T18:32:33+1100 [INFO] [api_server:1] Application startup complete.
```

As the logs show, the HTTP (not HTTPS) server is started.

### Expected behavior

I would expect the HTTPS server to be started as would happen when providing the same SSL options on the command line:

```bash
$ bentoml serve service:svc --production --ssl-certfile keys/cert.crt --ssl-keyfile keys/key.key --verbose
2023-02-15T18:33:27+1100 [DEBUG] [cli] Importing service "service:svc" from working dir: "/home/ryan/whatever/bentoml/src"
2023-02-15T18:33:28+1100 [DEBUG] [cli] 'transcription_pipeline' imported from source: bentoml.Service(name="transcription_pipeline", import_str="service:svc", working_dir="/home/ryan/whatever/bentoml/src")
2023-02-15T18:33:28+1100 [DEBUG] [cli] Runner map: {}
2023-02-15T18:33:28+1100 [INFO] [cli] Prometheus metrics for HTTPS BentoServer from "service:svc" can be accessed at https://localhost:3000/metrics.
2023-02-15T18:33:28+1100 [INFO] [cli] Installing handle_callback_exception to loop
2023-02-15T18:33:28+1100 [INFO] [cli] Registering signals...
2023-02-15T18:33:29+1100 [INFO] [cli] Starting master on pid 2605076
2023-02-15T18:33:29+1100 [DEBUG] [cli] Socket bound at 0.0.0.0:3000 - fd: 3
2023-02-15T18:33:29+1100 [INFO] [cli] sockets started
2023-02-15T18:33:29+1100 [DEBUG] [cli] Initializing watchers
2023-02-15T18:33:29+1100 [DEBUG] [cli] cmd: /usr/bin/python
2023-02-15T18:33:29+1100 [DEBUG] [cli] args: ['-m', 'bentoml_cli.worker.http_api_server', 'service:svc', '--fd', '$(circus.sockets._bento_api_server)', '--runner-map', '{}', '--working-dir', '/home/ryan/whatever/bentoml/src', '--backlog', '2048', '--worker-id', '$(CIRCUS.WID)', '--prometheus-dir', '/home/ryan/bentoml/prometheus_multiproc_dir', '--ssl-certfile', 'keys/cert.crt', '--ssl-keyfile', 'keys/key.key']
2023-02-15T18:33:29+1100 [DEBUG] [cli] process args: ['/usr/bin/python', '-m', 'bentoml_cli.worker.http_api_server', 'service:svc', '--fd', '3', '--runner-map', '{}', '--working-dir', '/home/ryan/whatever/bentoml/src', '--backlog', '2048', '--worker-id', '1', '--prometheus-dir', '/home/ryan/bentoml/prometheus_multiproc_dir', '--ssl-certfile', 'keys/cert.crt', '--ssl-keyfile', 'keys/key.key']
2023-02-15T18:33:29+1100 [DEBUG] [cli] running api_server process [pid 2605142]
2023-02-15T18:33:29+1100 [INFO] [cli] Arbiter now waiting for commands
2023-02-15T18:33:29+1100 [INFO] [cli] api_server started
2023-02-15T18:33:29+1100 [INFO] [cli] Starting production HTTPS BentoServer from "service:svc" listening on https://0.0.0.0:3000 (Press CTRL+C to quit)
2023-02-15T18:33:29+1100 [DEBUG] [api_server:1] Importing service "service:svc" from working dir: "/home/ryan/whatever/bentoml/src"
2023-02-15T18:33:31+1100 [DEBUG] [api_server:1] 'transcription_pipeline' imported from source: bentoml.Service(name="transcription_pipeline", import_str="service:svc", working_dir="/home/ryan/whatever/bentoml/src")
2023-02-15T18:33:31+1100 [DEBUG] [api_server:1] 'tracing.sample_rate' is set to zero. No traces will be collected. Please refer to https://docs.bentoml.org/en/latest/guides/tracing.html for more details.
2023-02-15T18:33:31+1100 [INFO] [api_server:1] Started server process [2605142]
2023-02-15T18:33:31+1100 [INFO] [api_server:1] Waiting for application startup.
2023-02-15T18:33:31+1100 [INFO] [api_server:1] Application startup complete.
```

I also confirmed that bentoml was finding and parsing the config file because when temporarily inserting an invalid value in another section, bentoml would report the error. So it is just specifically the "ssl" section that is ignored. E.g. Adding this section successfully detects the error:

```yaml
runners:
  timeout: 100xyz
```

Although the command line arguments work, I rely on the configuration file in order to run from a cloud deployment.

#### System information

`bentoml`: 1.0.14
`python`: 3.10.9
`platform`: Linux-6.1.8-arch1-1-x86_64-with-glibc2.36

**Comment by aarnphm** [comment]
This bug has been reproduced and will be addressed shortly. This has to do with the value from CLI is overriding the default arguments passed via the Container configuration.

This will be addressed by #3597

**Comment by sauyon** [comment]
I believe this has been released and fixed, closing. Feel free to reopen if this isn't resolved.
